File: sampleandroid/src/main/java/com/example/sampleandroid/model/DbSimpleScreen.kt
package com.example.sampleandroid.model

import androidx.compose.Composable
import androidx.compose.mutableStateOf
import androidx.ui.core.Modifier
import androidx.ui.foundation.Text
import androidx.ui.layout.Column
import androidx.ui.layout.Row
import androidx.ui.material.Button
import com.example.sampleandroid.R
import kim.jeonghyeon.androidlibrary.compose.data
import kim.jeonghyeon.androidlibrary.compose.resourceStateOf
import kim.jeonghyeon.androidlibrary.compose.widget.TextField
import kim.jeonghyeon.androidlibrary.compose.widget.VerticalListView
import kim.jeonghyeon.androidlibrary.extension.resourceToString
import kim.jeonghyeon.pergist.asListFlow
import kim.jeonghyeon.sample.Word
import kim.jeonghyeon.sample.WordQueries
import kim.jeonghyeon.sample.di.serviceLocator

class DbSimpleScreen(private val wordQueries: WordQueries = serviceLocator.wordQueries) : ModelScreen() {
    override val title: String = R.string.db_simple.resourceToString()

    private val wordList by lazy { resourceStateOf<List<Word>>() }
    private val newWord by lazy { mutableStateOf("") }

    override fun initialize() {
        wordList.load(initStatus, wordQueries.selectAll().asListFlow())
    }

    private fun onClickAdd() {
        wordQueries.insert(newWord.value)
    }

    @Composable
    override fun view() {

        //todo check if job is cancelled if composable leave
        Column {
            VerticalListView(wordList.data(), Modifier.weight(1f)) {
                Text(it.toString())
            }
            Row {
                TextField(newWord)
                Button(onClick = ::onClickAdd) {
                    Text(R.string.add.resourceToString())
                }
            }
        }


    }

    @Composable
    override fun compose() {
        super.compose()
    }


}

File: sample/sample-native/kotlinIOS/screen/model/DbSimpleScreen.swift
//
// Created by hyun kim on 20/07/20.
// Copyright (c) 2020 hyun kim. All rights reserved.
//

import Foundation

import SwiftUI
import sample_base

func DbSimpleScreen(_ model: DbSimpleViewModel) -> some View {
    Screen(model) {
        Column {
            List(+model.wordList as? [Word] ?? [Word](), id: \.self.id) { item in
                Text("id : \(item.id), text : \(item.text)")
            }
            Row {
                SampleTextField("Enter value", model.newWord)
                Button("Add") {
                    model.onClickAdd()
                }
            }
        }
    }
}
